### ISSUE TITLE
Commit 57350a3 breaks my private layer

### ISSUE THREAD

**Issue by bixuanzju** [syl20bnr/spacemacs#6381]
#### Description :octocat:
Commit 57350a3 breaks my private layer by deleting required packages.

Specifically I have a proof-general layer where I declare the `company-coq` package as follows:

```elisp
(setq proof-general-packages
      '(
        (proof-general :location local)
        company-coq
        ))

(when (configuration-layer/package-usedp 'company)
  (defun proof-general/init-company-coq ()
    (use-package company-coq
      :defer t
      :init (add-hook 'coq-mode-hook #'company-coq-mode)
      :config
      (progn
        (spacemacs/set-leader-keys-for-major-mode 'coq-mode
          "g" 'company-coq-jump-to-definition
          "d" 'company-coq-doc
          "o" 'company-coq-occur)))))
```

I can confirm before this commit, my layer worked.

#### Reproduction guide :beetle:
- Start Emacs
- And behold

*Observed behaviour:* :eyes: :broken_heart:
`company-coq` is deleted. Strangely if I run `dotspacemacs/sync-configuration-layers` then `company-coq` is re-installed. But restarting Emacs would again delete that package :(

*Expected behaviour:* :heart: :smile:
My layers works as expected (every declared package installed).

#### System Info :computer:
- OS: darwin
- Emacs: 25.0.95.1
- Spacemacs: 0.105.21
- Spacemacs branch: nil (rev. 57350a3)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm
 (auto-completion :variables auto-completion-enable-sort-by-usage t)
 emacs-lisp
 (git :variables git-magit-status-fullscreen t)
 (markdown :variables markdown-live-preview-engine 'vmd)
 org
 (shell :variables shell-default-height 30 shell-default-position 'bottom)
 spell-checking syntax-checking version-control osx colors idris
 (haskell :variables haskell-completion-backend 'intero haskell-enable-hindent-style "johan-tibell")
 latex
 (evil-snipe :variables evil-snipe-enable-alternate-f-and-t-behaviors t)
 ocaml twitter games lua sml proof-general)
```

**Comment by syl20bnr** [comment]
I see, I imagine the second pass to load `packages.el` is broken.
It is not recommended to use `(when configuration-layer/package-usedp 'company)` since it schedules a second pass on `packages.el` to resolve the when clause (which seems to be broken).
The workaround and recommended way to add the condition is with `:toggle` keyword:

```elisp
(setq proof-general-packages
      '(
        (proof-general :location local)
        (company-coq :toggle (configuration-layer/package-usedp 'company))
        ))
``` 

Should do it until I fixed the bug, thanks for the report.

**Comment by bixuanzju** [comment]
@syl20bnr Thank you, the workaround works, I will stick with that for now.

**Comment by syl20bnr** [comment]
I found the issue, will fix it tomorrow :-)

**Comment by syl20bnr** [comment]
@bixuanzju did I fix this or not ? Sorry for the late reply.

**Comment by bixuanzju** [comment]
Now I use the approach you recommended above, no issue so far.

**Comment by syl20bnr** [comment]
Ok, I checked the code and for me it seems to be fixed, anyway I plan to remove this hack at some point.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
